profiling/src/profile/mod.rs

```diff
@@ -438,62 +274,21 @@ impl<'a> ProfileBuilder<'a> {
     }
 }
 
-impl<'a> Default for ProfileBuilder<'a> {
-    fn default() -> Self {
-        Self::new()
-    }
+trait Dedup<T: Item> {
+    /// Deduplicate the Item and return its associated Id.
+    /// # Panics
+    /// Panics if the number of items overflows the storage capabilities of
+    /// the associated Id type.
+    fn dedup(&mut self, item: T) -> <T as Item>::Id;
 }
 
-trait DedupExt<T: Eq + Hash> {
-    fn dedup(&mut self, item: T) -> usize;
-
-    fn dedup_ref<'a, Q>(&mut self, item: &'a Q) -> usize
-    where
-        T: Eq + Hash + From<&'a Q> + Borrow<Q>,
-        Q: Eq + Hash + ?Sized;
-}
-
-impl<T: Sized + Hash + Eq> DedupExt<T> for FxIndexSet<T> {
-    fn dedup(&mut self, item: T) -> usize {
+impl<T: Item> Dedup<T> for FxIndexSet<T> {
+    fn dedup(&mut self, item: T) -> <T as Item>::Id {
         let (id, _) = self.insert_full(item);
```

Comment:
Do the same assertion as `intern` does

Reply:
I don't think it makes sense to in this case: if it already exists, that's fine. The string case is different because it can't do a simple `insert_full` because it has to deal with `&str` vs `String` differences.